### DISCUSSION TITLE
3.9.x

### DISCUSSION THREAD

**Discussion by mokun** [mars-sim/mars-sim#1504] [General]
@bevans2000 

What major things left to do for 3.9.0 ?

**Comment by bevans2000** [comment]
> @bevans2000 
> 
> What major things left to do for 3.9.0 ?

I have some changes in a branch pending for #1197

**Reply by bevans2000** [reply]
I think #1236 would be a good one to include. I want to look at this one next

**Comment by bevans2000** [comment]
@mokun I think we should look to close down 3.9.0 in February. It will be 6 months since 3.8.0.
I would like this to be aligned with the completion of the redesigned web site.
The only thing I think needs doing to the code is to reduce the output log, seems to have grown.

Also we should move to 3.10.0 next as there is nothing significant planned to justify a new major version.

**Reply by bevans2000** [reply]
I'm focusing on getting the new web site done

**Comment by mokun** [comment]
Now, I still worry about https://github.com/mars-sim/mars-sim/issues/1513 and we should probably offer a quick fix on that prior to releasing 3.9.0.

**Reply by bevans2000** [reply]
Fair enough. Just added some comment to that Issue.

**Comment by bevans2000** [comment]
I have added quite a bit of change for the methane depletion problem. You might need to run a longer simulation.
At weekend I will reset the version number to 3.9.0 and dropped the pre in preparation. This will allow me to run a generation of the website on ages with the latest version

**Reply by mokun** [reply]
I can think of at least the following changes : 

1. Annotate all the changes under a new version tag called 3.9.0 in CHANGELOG.md, whatsnews.html, version_history.html
2. Update the year tag from 2024 to 2025 on all md files and various files
3. Update all pom.xml to 3.9.0

**Reply by bevans2000** [reply]
Fine. There is a wiki page explaining the release steps. I don't think it includes updating the year, I'll check at weekend

**Reply by bevans2000** [reply]
So why do we have CHANGELOG.md, whatsnews.html, version_history.html, they all seem to contain the same information. I think we should just have a single list.
Also I want to tweak the Create Release GitHub action once we are ready

**Reply by mokun** [reply]
Yay. We can consolidate CHANGELOG.md and version_history.html into just one file in future. 

For now, there are links that load whatsnews.html and version_history.html separately. 

whatsnews.html is different from version_history.html obviously as whatsnews.html only records the changes at the current working version.

**Reply by bevans2000** [reply]
It would work if we just used the version history file and not the what's new. We are duplicating the description of the current release in both

**Comment by bevans2000** [comment]
Going to close off release 3.9.0 at weekend. I think it's as complete as it's going to be.

**Reply by bevans2000** [reply]
Done
